### DISCUSSION TITLE
OC 4.x > Documentation

### DISCUSSION THREAD

**Discussion by osworx** [opencart/opencart#10954] [General]
Badly missing the **promised Documentations**!
see: https://github.com/opencart/opencart/issues/10922#issuecomment-1140285134

Because of the issues installing languages, themes, extensions in OC 4.x, it would be time to finish them (have you already started Daniel ?).
Otherwise I do really see no future for the 4.x release and any "extension" development for it.

**Comment by felipoantonoff** [comment]
Good suggestion had some more complex changes in Opencart 4, it would be nice maybe some simple guide showing how to convert XMLs (OCMODs) and types of ways to use Events correctly.

An example, I have XMLs that change controller, model or library and view lines, but I don't know how I can migrate them to Events or another tool, since the View part I have ideas because the forum has already given suggestions, but I believe they are current forms a bit crude and that will degrade performance and probably cause conflicts that are difficult to parse, so it would be interesting examples even maybe taking a standard XML and converting it to Events.

The new structure and location of the Modules is also an interesting summary.

**Comment by mhcwebdesign** [comment]
@osworx: I agree, we need some documentation! I was trying to port our Export/Import tool to OpenCart 4, using the following folder structure:

extension/export_import/admin/controller/tool/export_import.php
extension/export_import/admin/language/en-gb/tool/export_import.php
extension/export_import/admin/model/tool/export_import.php
extension/export_import/admin/view/stylesheet/export_import.css
extension/export_import/admin/view/template/export_import.twig
extension/export_import/system/library/export_import/*
extension/export_import/install.json

I also noticed the namespaces don't match up with the underlying folder structure. For example, the 

extension/opencart/admin/controller/module/account.php

uses the namespace

Opencart\Admin\Controller\Extension\Opencart\Module

instead of the expected

Opencart\Extension\Opencart\Admin\Controller\Module

This is totally confusing and goes against common practices in namespace usage in programming languages such as PHP, Java, and others!
